src/tbb/task_group_context.cpp

```diff
@@ -240,19 +243,15 @@ bool task_group_context_impl::is_group_execution_cancelled(const d1::task_group_
     return ctx.my_cancellation_requested.load(std::memory_order_relaxed) != 0;
 }
 
-// IMPORTANT: It is assumed that this method is not used concurrently!
+// Thread-safe reset implementation using compare_exchange
```

Comment:
The new comment is misleading, as there is no CAS on this path anymore.

And I think the old comment has some value. Even if the method will allow concurrent invocations, it still should not be used while there are tasks in the context, as it can erase the cancellation signal.